lib/src/license_detection/normalizer_test.dart

```diff
@@ -0,0 +1,32 @@
+import 'normalizer.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('Normalizer tests', () {
```

Comment:
I think we can make two groups of tests here:

(A) A group where we have one or more test cases for each guideline:
```dart
group('normalize implements', () {
  test('SPDX guideline 4.1.1', () {
    expect(normalize('Hello World'), equals('hello world'));
  });

   // TODO: write a test case for each guideline, keep these test cases as small as possible
   ....
});

final _testVectors = {
  'Hello World this is a complciated test case': 'hello world this is a complciated test case',
}

group('normalize test vectors', () {
  for (final key in _testVectors.keys) {
     test(key, () {.....
  }
});
```